--- src/google/protobuf/compiler/csharp/csharp_field_base.h
+++ src/google/protobuf/compiler/csharp/csharp_field_base.h
@@ -16,7 +16,7 @@
 #ifndef GOOGLE_PROTOBUF_COMPILER_CSHARP_FIELD_BASE_H__
 #define GOOGLE_PROTOBUF_COMPILER_CSHARP_FIELD_BASE_H__
 
-#include "google/protobuf/compiler/csharp/csharp_source_generator_base.h"
+#include "csharp_source_generator_base.h"
 #include "google/protobuf/compiler/code_generator.h"
 #include "google/protobuf/descriptor.h"
 #include "google/protobuf/io/printer.h"
